crates/json-rpc/src/response/error.rs

```diff
@@ -81,9 +81,12 @@ fn spelunk_revert(value: &Value) -> Option<Bytes> {
     }
 }
 
-impl<ErrData> fmt::Display for ErrorPayload<ErrData> {
+impl<ErrData: Default + fmt::Display+ Clone > fmt::Display for ErrorPayload<ErrData> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "error code {}: {}", self.code, self.message)
+        write!(f, "error code {}: {}, data: {}", self.code, self.message,
+               <Option<ErrData> as Clone>::clone(&self.data).
+                   unwrap_or_else(|| ErrData::default()).to_string().trim_matches('"')
+        )
```

Comment:
logging data makes sense and restricting ErrData to display makes sense I believe.

but this can be solved without out the default and the clone impl by mapping data, converting to string, if none this then should be an empty string

Reply:
change to:
```
impl<ErrData: fmt::Display> fmt::Display for ErrorPayload<ErrData> {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(
            f,
            "error code {}: {}, data: {}",
            self.code,
            self.message,
            self.data
                .as_ref()
                .map(|data| data.to_string())
                .unwrap_or_default()
        )
    }
}
```